src/Moryx.CommandCenter.Web/src/common/components/Menu/RoutingMenuItem.tsx

```diff
@@ -19,54 +18,42 @@ interface MenuItemState {
     IsOpened: boolean;
 }
 
-class RoutingMenuItem extends React.Component<RouteComponentProps<{}> & MenuItemProps, MenuItemState> {
-    private unregisterListenerCallback: UnregisterCallback;
+function RoutingMenuItem(props: MenuItemProps) {
+    const location = useLocation();
+    const navigate = useNavigate();
 
-    constructor(props: RouteComponentProps<{}> & MenuItemProps) {
-        super(props);
-        this.state = { IsOpened: this.isOpened(this.props.location) };
+    const isOpened = (location: any): boolean => {
+        return location.pathname.endsWith(`/${props.MenuItem.NavPath}`);
```

Comment:
So this should answer couple of your questions :)

I refer to [this](https://reactrouter.com/en/main/upgrading/v5#relative-routes-and-links) part of the migration guide.

> A few important things to notice about v6 in this example: <Route path> and <Link to> are **relative** ...

So if you were already on `.../CommandCenter#/databases` and clicked on a route with `path="/databases"` in v6 you would now land on `.../CommandCenter#/databases/databases`.
So I removed this prefix (same with modules). 

Problem is the previous `isOpen` and `isActive` checks used this prefix to decide if we are currently on `database/TestModule` or `database/DependentTestModule` for example. But now we just have `DependentTestModule` and `TestModule` without this prefix. A simple `endsWith("TestModule")` call returns `true` for both (same with `includes`). And you could think of module names where `startsWith` would return `true` in multiple cases. This resulted in multiple modules being shown as active in the module list.

To fix this I first assumed no module or database contains the forward slash in its name (I also think that would mess up the navigation of the site). So I concatenate slash with the module or database name and checked if the current location ends with this string. Now either `.../CommandCenter#/modules/TestModule` or `.../CommandCenter#/modules/DependentTestModule` can be active.

Reply:
Now that I think of it, navigating to  `.../CommandCenter#/modules/ResourceManager/console` removes the highlight in the list because the location does not end with `/ResourceManager`. So if should be `includes` instead. But now I assume now module or database is ever called "console" or "databases" for example

Reply:
I think we can assume that for now, if that topic comes up at some point we can try to resolve it 😊

Reply:
I added the prefix to the `NavPath` again and removed the first occurence in `<Route>` by doing `path={menuItem.NavPath.replace("/modules/", "")`. Turned `endsWith` into `startsWith`